packages/protocol/contracts/governance/LockedGold.sol

```diff
@@ -5,777 +5,438 @@ import "openzeppelin-solidity/contracts/math/SafeMath.sol";
-    address _account,
-    uint256 noticePeriod
+  function setAccountMustMaintain(
+    address account,
+    uint256 value,
+    uint256 timestamp
   )
-    external
-    view
-    returns (uint256, uint256)
+    public
+    onlyRegisteredContract(VALIDATORS_REGISTRY_ID)
+    nonReentrant
+    returns (bool)
   {
-    Account storage account = accounts[_account];
-    Commitment storage locked = account.commitments.locked[noticePeriod];
-    return (locked.value, locked.index);
+    accounts[account].balances.requirements = MustMaintain(value, timestamp);
+    emit AccountMustMaintainSet(account, value, timestamp);
   }
 
+  // TODO(asa): Dedup
   /**
-   * @notice Returns the value and index of a specified notified commitment.
-   * @param _account The address of the account.
-   * @param availabilityTime The availability time of the notified commitment.
-   * @return The value and index of the specified notified commitment.
+   * @notice Returns the account associated with `accountOrVoter`.
+   * @param accountOrVoter The address of the account or authorized voter.
+   * @dev Fails if the `accountOrVoter` is not an account or authorized voter.
+   * @return The associated account.
    */
-  function getNotifiedCommitment(
-    address _account,
-    uint256 availabilityTime
-  )
-    external
-    view
-    returns (uint256, uint256)
-  {
-    Account storage account = accounts[_account];
-    Commitment storage notified = account.commitments.notified[availabilityTime];
-    return (notified.value, notified.index);
+  function getAccountFromVoter(address accountOrVoter) external view returns (address) {
+    address authorizingAccount = authorizedBy[accountOrVoter];
+    if (authorizingAccount != address(0)) {
+      require(accounts[authorizingAccount].authorizations.voting == accountOrVoter);
+      return authorizingAccount;
+    } else {
+      require(isAccount(accountOrVoter));
+      return accountOrVoter;
+    }
   }
 
   /**
-   * @notice Returns the account associated with the provided delegate and role.
-   * @param accountOrDelegate The address of the account or voting delegate.
-   * @param role The delegate role to query for.
-   * @dev Fails if the `accountOrDelegate` is a non-voting delegate.
-   * @return The associated account.
+   * @notice Returns the total amount of locked gold in the system.
+   * @return The total amount of locked gold in the system.
    */
-  function getAccountFromDelegateAndRole(
-    address accountOrDelegate,
-    DelegateRole role
-  )
-    public
-    view
-    returns (address)
-  {
-    address delegatingAccount = delegations[accountOrDelegate];
-    if (delegatingAccount != address(0)) {
-      require(accounts[delegatingAccount].delegates[uint256(role)] == accountOrDelegate);
-      return delegatingAccount;
-    } else {
-      return accountOrDelegate;
-    }
+  function getTotalLockedGold() external view returns (uint256) {
+    return totalNonvoting.add(getElection().getTotalVotes());
```

Comment:
not important, but weird name: sum of `gold + votes`

Reply:
again.. not that important... not sure of a better name...
"getTotalVotingGold"???